apollo-portal/src/main/resources/static/scripts/directive/delete-namespace-modal-directive.js

```diff
@@ -39,24 +37,18 @@ function deleteNamespaceModalDirective($window, $q, $translate, toastr, AppUtil,
                 //1. check operator has master permission
                 checkPermission(toDeleteNamespace).then(function () {
 
-                    //2. check namespace's master branch has not instances
-                    if (!checkMasterInstance(toDeleteNamespace)) {
-                        return;
-                    }
-
-                    //3. check namespace's gray branch has not instances
-                    if (!checkBranchInstance(toDeleteNamespace)) {
-                        return;
-                    }
+                    if(toDeleteNamespace.isLinkedNamespace){
```

Comment:
For public and private namespaces, I think we still need to check the instances by`checkMasterInstance` and `checkBranchInstance`?

Reply:
- Here it is considered that the user delete action is expected.
- Only the current environment and the usage of the current cluster can be checked here, and deleting the AppNamespace will affect all clusters, so the logic here remains consistent with the administrator's deletion
- Or, is there a need to check the usage of all clusters in all environments?

Reply:
In this case, there is a detailed message indicating that this operation will cause the instances unable to get the configuration, so I think it's ok to not `checkMasterInstance` and `checkBranchInstance`.
If we don't check for public and private namespaces, I suppose we don't need to check for linked namespaces?

Reply:
> If we don't check for public and private namespaces, I suppose we don't need to check for linked namespaces?

Yes, I think they are the same type of problem

Reply:
So maybe we could simplify the logic as the following?
```javascript
                //1. check operator has master permission
                checkPermission(toDeleteNamespace).then(function () {

                    if (!toDeleteNamespace.isPublic || toDeleteNamespace.isLinkedNamespace) {
                        showDeleteNamespaceConfirmDialog();
                    } else {
                        //2. check public namespace has not associated namespace
                        checkPublicNamespace(toDeleteNamespace).then(function () {
                            showDeleteNamespaceConfirmDialog();
                        });
                    }
                })
```

Reply:
The logic for deleting different types of Namespace on the user side has been unified, and a uniform instance usage prompt message is added before deletion, e.g.

- Delete private Namespace
<img width="992" alt="image" src="https://user-images.githubusercontent.com/18591662/184076986-f7d1d89b-52cb-46ac-b141-2cdb9034cebe.png">
<img width="981" alt="image" src="https://user-images.githubusercontent.com/18591662/184077101-fce82c35-58dd-4b74-ae50-3f917be3eff5.png">

-   Delete public Namespace

<img width="979" alt="image" src="https://user-images.githubusercontent.com/18591662/184077461-6c998138-8739-4ef0-9132-922ebd630c03.png">
<img width="969" alt="image" src="https://user-images.githubusercontent.com/18591662/184077664-751cd444-4c15-4fa0-8366-19054dd404b7.png">

-  Delete linked Namespace
<img width="965" alt="image" src="https://user-images.githubusercontent.com/18591662/184078030-a18a9380-ef5d-4706-a602-9240f0a315f8.png">
<img width="974" alt="image" src="https://user-images.githubusercontent.com/18591662/184078102-9a6b7ffc-43ee-4a79-98f6-844642712900.png">